### PR TITLE
Bmo deep dive presentation

### PR DESCRIPTION
**Pull request by hroyrh**
A presentation to cover the BMO in detail.

### CODE REVIEW

**Comment by kashifest** [comment]
/hold
Until default branch of the repo is main

**Comment by jklippel** [comment]
This gives a lot of insight. I see the PR is on hold due to the default-branch name change. It was changed in the meantime.
Is there any other reason this is still on hold?

It is a really helpful piece of documentation that is currently hidden here.

File: docs/presentations/bmo-deep-dive/bmo-deep-dive.html [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+			<!-- Theme used for syntax highlighted code -->
+			<link rel="stylesheet" href="plugin/highlight/monokai.css" id="highlight-theme">
+		</head>
+		<body>
+			<div class="reveal">
+				<div class="slides"
+				     style="font-size:50%;text-align:left">
+					<section data-markdown
+					         data-separator-vertical="^\n\n\n"
+									 data-background-image="https://github.com/metal3-io/metal3-docs/raw/master/images/metal3.png"
```
Comment (furkatgofurov7) [inline docs/presentations/bmo-deep-dive/bmo-deep-dive.html:23]: after branch switch (master ==> main), we need to update this and all other BMO reference links in the document accordingly

File: docs/presentations/bmo-deep-dive/bmo-deep-dive.html [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+<br>
+<br>
+Then we will take a look at sections of a BareMetalHost Object after its creation like Spec, Status, etc.
+<br>
+<br>
+For further details about the BMO API, refer the [documentation here](https://github.com/metal3-io/baremetal-operator/blob/master/docs/api.md).
+
+
+
+### Sample BareMetalHost Definition
```
Comment (furkatgofurov7) [inline docs/presentations/bmo-deep-dive/bmo-deep-dive.html:96]:
```suggestion
### Sample BareMetalHost Custom Resource Definition
```

File: docs/presentations/bmo-deep-dive/bmo-deep-dive.html [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bmo-master-0-bmc-secret
```
Comment (furkatgofurov7) [inline docs/presentations/bmo-deep-dive/bmo-deep-dive.html:102]: This does not match with the secret name provided in BMH later, it should be 
```suggestion
  name: bm0-bmc-secret
```

File: docs/presentations/bmo-deep-dive/bmo-deep-dive.html [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+<br>
+<br>
+
+For more details refer [this documentation](https://github.com/metal3-io/baremetal-operator/blob/master/docs/ironic-endpoint-keepalived-configuration.md).
+
+
+
+
+
+# Roadmap
```
Comment (furkatgofurov7) [inline docs/presentations/bmo-deep-dive/bmo-deep-dive.html:271]: Do we have one which is up-to-date?

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**